pkg/controller/controller_binding.go

```diff
@@ -365,13 +365,13 @@ func (c *controller) injectBinding(binding *v1alpha1.Binding, credentials map[st
 	}
 
 	found := false
-
 	_, err := c.kubeClient.Core().Secrets(binding.Namespace).Get(binding.Spec.SecretName, metav1.GetOptions{})
 	if err == nil {
 		found = true
 	}
 
 	if found {
+		// TODO: Check ownerReferences first?
 		_, err = c.kubeClient.Core().Secrets(binding.Namespace).Update(secret)
 	} else {
 		_, err = c.kubeClient.Core().Secrets(binding.Namespace).Create(secret)
```

Comment:
Agreed, but OSB requires to execute only one operation for service at a time (I haven't checked the code with exclusive lock myself, but it should be somewhere), so the problem with concurrent updates exists only if there are 2 Bindings pointing to the same secret, for example.
Store is a bit early to add there, since there are no Secret watchers yet. I would prefer to keep the code as simple as possible, but rock solid at the same time.